lib.rs

```diff
@@ -28,6 +30,27 @@ pub trait FallibleVec<T> {
     fn try_extend_from_slice(&mut self, other: &[T]) -> Result<(), ()> where T: Clone;
 }
 
+/// Reserves the upper limit of what `src` can generate before reading all
+/// bytes until EOF in this source, placing them into buf. If the allocation
+/// is unsuccessful, or reading from the source generates an error before
+/// reaching EOF, this will return an error. Otherwise, it will return the
+/// number of bytes read.
+///
+/// Since `src.limit()` may return a value greater than the number of bytes
+/// which can be read from the source, it's possible this function may fail
+/// in the allocation phase even though allocating the number of bytes available
+/// to read would have succeeded. In general, it is assumed that the callers
+/// have accurate knowledge of the number of bytes of interest and have created
+/// `src` accordingly.
+pub fn try_read_to_end<T>(src: &mut std::io::Take<T>, buf: &mut Vec<u8>) -> Result<usize, ()>
```

Comment:
This looks like a fine approach, but I think it should live inside mp4parse-rust (in the same way allocate_read_buf does), since it's built on FallibleVec::try_reserve and this crate is trying to provide the minimum possible in terms of implementation - ideally we will be able to drop this crate and use Rust's fallible allocations directly.